src/validation.h

```diff
@@ -862,6 +862,10 @@ class ChainstateManager
         CBlockIndex** ppindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
     friend CChainState;
 
+    // Returns nullptr if no snapshot ahs been loaded.
+    CBlockIndex* getSnapshotBaseBlock() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
+    std::optional<int> getSnapshotHeight() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
In commit "add ChainstateManager.getSnapshot{Height,BaseBlock}()" (d8f32839cfbc8cbfe194d190bdc19fb3c4d6e32a)

Would rename `getSnapshotHeight` to `getSnapshotBaseHeight` for two reasons:

- To be consistent with `getSnapshotBaseBlock`
- To disambiguate. "Snapshot" could refer to snapshot chainstate rather than snapshot itself, and chainstate may have a height higher than the initial snapshot

This is a tangent, but I'm increasingly disliking the terms "ibd chainstate" and "snapshot chainstate." If had a magic wand I would abolish them everywhere. I think ideally a chainstate would consist of a UTXO set, a tip pointer, and "validated starting from" block pointer and "validated starting from" UTXO hash. ChainstateManager would have a list of chainstates, and one of them would be active with rest inactive. Network code would download blocks to validate chainstates, prioritizing blocks for the active chainstate. ChainstateManager would connect the incoming blocks, and if the tip of any chainstate crossed the "validated starting from" block of a later chainstate, and UTXO hash matched, the later "validated_starting_from" pointer would moved back to the earlier one, and earlier chainstate (optionally) discarded.
Probably having explicit tip and "validated starting from" pointers would also allow CBlockIndex validation flags to be simplified so more code doesn't need to care about assumeutxo. This could all be cleanup though. More important to get the existing code that's already merged functional and not change things at this point.

Reply:
Renamed. Agree with your thoughts on the better design, and also agree that the path from here to there is not so long. I'd just prefer to wait until we actually need to revamp the design (e.g. multiple parallel background chainstates) to do all the code changes.